packages/directlinespeech/__tests__/utilities/fetchSpeechCredentialsWithCache.js

```diff
@@ -32,19 +45,44 @@ async function fromSubscriptionKey({ region, subscriptionKey }) {
   };
 }
 
+async function getDirectLineTokenFromSecret(channelSecret) {
+  const bearerToken = "Bearer " + channelSecret
+  const res = await fetch("https://directline.botframework.com/v3/directline/tokens/generate", {
+    headers: {
+      'Authorization': bearerToken
+    },
+    method: 'POST'
+  });
+
+  if (!res.ok) {
+    throw new Error(`Failed to fetch authorization token for Direct Line, server returned ${res.status}`);
+  }
+
+  const { token: directLineToken } = await res.json();
+  return directLineToken;
+}
+
 let credentialsPromise;
 
-export default async function fetchSpeechCredentialsWithCache() {
+export default async function fetchSpeechCredentialsWithCache(enableInternalHttpSupport = false) {
   if (!credentialsPromise) {
-    const { SPEECH_SERVICES_REGION, SPEECH_SERVICES_SUBSCRIPTION_KEY } = process.env;
-
+    const { SPEECH_SERVICES_REGION, SPEECH_SERVICES_SUBSCRIPTION_KEY, SPEECH_SERVICES_DIRECT_LINE_SECRET } = process.env;
     if (SPEECH_SERVICES_REGION && SPEECH_SERVICES_SUBSCRIPTION_KEY) {
       credentialsPromise = fromSubscriptionKey({
         region: SPEECH_SERVICES_REGION,
         subscriptionKey: SPEECH_SERVICES_SUBSCRIPTION_KEY
       });
+      
+      if (enableInternalHttpSupport && SPEECH_SERVICES_DIRECT_LINE_SECRET) {
+        const { authorizationToken, region } = await credentialsPromise;
+        const directLineToken = await getDirectLineTokenFromSecret(SPEECH_SERVICES_DIRECTLINE_SECRET);
+        return {authorizationToken, region, directLineToken };
```

Comment:
Cleanliness nitpick: alphabetize.
```suggestion
        return {authorizationToken, directLineToken, region };
```